pkg/driver/controller.go

```diff
@@ -354,27 +354,20 @@ func (d *controllerService) ControllerPublishVolume(ctx context.Context, req *cs
 	volumeID := req.GetVolumeId()
 	nodeID := req.GetNodeId()
 
-	if !d.cloud.IsExistInstance(ctx, nodeID) {
-		return nil, status.Errorf(codes.NotFound, "Instance %q not found", nodeID)
-	}
-	disk, err := d.cloud.GetDiskByID(ctx, volumeID)
-	if err != nil {
-		if errors.Is(err, cloud.ErrNotFound) {
-			return nil, status.Error(codes.NotFound, "Volume not found")
-		}
-		return nil, status.Errorf(codes.Internal, "Could not get volume with ID %q: %v", volumeID, err)
+	if !d.inFlight.Insert(volumeID) {
+		return nil, status.Error(codes.Aborted, fmt.Sprintf(internal.VolumeOperationAlreadyExistsErrorMsg, volumeID))
 	}
+	defer d.inFlight.Delete(volumeID)
 
-	// If given volumeId already assigned to given node, will directly return current device path
+	klog.V(2).InfoS("ControllerPublishVolume: attaching", "volumeID", volumeID, "nodeID", nodeID)
 	devicePath, err := d.cloud.AttachDisk(ctx, volumeID, nodeID)
 	if err != nil {
 		if errors.Is(err, cloud.ErrVolumeInUse) {
-			return nil, status.Error(codes.FailedPrecondition, strings.Join(disk.Attachments, ","))
+			return nil, status.Errorf(codes.FailedPrecondition, "Volume %q is already attached to a different node", volumeID)
```

Comment:
Do we need to include node id in the error message so we know which node was requested to have the volume attach to?

Reply:
We log that information right before calling `AttachDisk`: https://github.com/torredil/aws-ebs-csi-driver/blob/62c55109900ee31e9d1b3c9d3e96feeabdda7714/pkg/driver/controller.go#L362 but more clarity is always helpful. Changed to `"Volume %q is already attached to a different node, expected node: %q", volumeID, nodeID)`.